File: daily/2026/3612_Process_String_with_Special_Operations_I.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    string processStr(string s) {
        string ans;
        for (char c : s){
            if (c == '*'){
                if (!ans.empty()) ans.pop_back();
            }
            else if (c == '#') ans += ans;
            else if (c == '%') reverse(ans.begin(), ans.end());
            else ans += c;
        }
        return ans;
    }
};

int main(){
    string s = "a#b%*";
    // string s = "z*#";
    Solution S;
    cout << S.processStr(s) << endl;
    return 0;
}

File: daily_python/3612_Process_String_with_Special_Operations_I.py
import os
import sys
from typing import List

class Solution:
    def processStr(self, s: str) -> str:
        ans = []
        for c in s:
            if c == '*':
                if c:
                    ans.pop()
            elif c == '#':
                ans += ans.copy()
            elif c == '%':
                ans = ans[::-1]
            else:
                ans.append(c)
        return "".join(ans)
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            s = lines[0].split('=')[1].strip()[1:-2]
            dataset.append(s)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for s in dataset:
        results.append(solution.processStr(s))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')